File: benchmarks/cases/series_pi/series_pi.cpp
#include <cstdio>

int main() {
    long n = 10000000;
    double sum = 0.0;
    double sign = 1.0;

    for (long k = 0; k < n; k++) {
        double denom = 2.0 * k + 1.0;
        sum = sum + sign / denom;
        sign = -sign;
    }

    double pi = sum * 4.0;
    printf("%lf\n", pi);
    return 0;
}

File: benchmarks/cases/series_pi/series_pi.rs
fn main() {
    let n: i64 = 10000000;
    let mut sum: f64 = 0.0;
    let mut sign: f64 = 1.0;

    for k in 0..n {
        let denom = 2.0 * k as f64 + 1.0;
        sum = sum + sign / denom;
        sign = -sign;
    }

    let pi = sum * 4.0;
    println!("{:.6}", pi);
}
